docs/rules/no-setup-in-describe.md

```diff
@@ -0,0 +1,71 @@
+
+# Disallow setup in describe blocks (no-setup-in-describe)
+
+Setup for test cases in mocha should be done in `before`, `beforeEach`, or `it` blocks. Unfortunately there is nothing stopping you from doing setup directly inside a `describe` block.
+
+```js
+describe('something', function () {
+    const a = setup();
+    const b = a.getter;
+    it('should work', function () {
+        assert(a === b);
+    });
+});
+```
+
+Any setup directly in a `describe` is run before all tests execute. This is undesirable primarily for two reasons:
+
+1. When doing TDD in a large codebase, all setup is run for tests that don't have `only` set. This can add a substantial amount of time per iteration.
+2. If global state is altered by the setup of another describe block, your test may be affected.
+
+This rule reports all function calls and use of the dot operator (due to getters and setters) directly in describe blocks.
+
+If you're using [dynamically generated tests](https://mochajs.org/#dynamically-generating-tests), you should disable this rule.
```

Comment:
Ah, I was just going to ask about this 😀 .
I think it is ok that this rule will only work for non dynamically generated tests.